Integrations/integration-MISP.yml

```diff
@@ -0,0 +1,534 @@
+                        }
+                        return res;
+                    }
+                    return transformSingleObj(obj);
+                };
+            }
+
+
+    function translateEvents(rawResponse, contextPath, title){
+        var returnObject = mapObjFunction(entitiesDict.event)(dq(rawResponse,'response.Event'));
+
+        var ec = {};
+            if (contextPath) {
+                ec[contextPath] = returnObject;
+            }
+            return {
+                Type: entryTypes.note,
+                ContentsFormat: formats.json,
+                Contents: rawResponse,
+                ReadableContentsFormat: formats.markdown,
+                HumanReadable: tableToMarkdown(title, returnObject),
+                EntryContext: ec
+            };
+    }
+
+
+
+
+
+
+
+    function getProvidedArgsForCommand(command, passedArgs) {
+        var currentArgs = {};
+        var allArgs = commands[command].args;
+            for (var j = 0; j < allArgs.length; j++){
+                if (passedArgs[allArgs[j]]){
+                  currentArgs[allArgs[j]] = passedArgs[allArgs[j]];
+                }
+            }
+        return currentArgs;
+    }
+
+    function getHashFormat(hash) {
+        switch(hash.length){
+            case 32:
+                return 'md5';
+            case 40:
+                return 'sha1';
+            case 64:
+                return 'sha256';
+            default:
+                throw "invalid hash length, please enter file hash of format md5, sha1 or sha256";
+        }
+    }
+
+    function getMISPThreatLevelStr(threat_level_id) {
+        switch(threat_level_id){
+            case '1':
+                return 'HIGH';
+            case '2':
+                return 'MEDIUM';
+            case '3':
+                return 'LOW';
+            case '4':
+                return 'UNDEFINE';
+            default:
+                return 'Invalid MISP Threat Level with threat_level_id: ' + threat_level_id;
+        }
+    }
+
+
+    function doGeneralPost(command, passedArgs){
+        var currentArgs = getProvidedArgsForCommand(command, passedArgs);
+        var requestUrl = params.url + commands[command].url;
+        var body = {'request': currentArgs };
+        var res = http(
+            requestUrl,
+            {
+                Method: 'POST', // Can be POST, PUT, DELETE, HEAD, OPTIONS or CONNECT
+                Headers: {
+                    'Authorization': [params.api_key],
+                    'Accept': ['application/json'],
+                    'Content-Type': ['application/json']
+                },
+                Body: JSON.stringify(body)
+            },
+            params.insecure,
+            params.useproxy
+        );
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+        return JSON.parse(res.Body);
```

Comment:
It might be good to check that the content type of the response is application/json.
we don't do it much now, but it's a good time to start